frontend/src/components/AddItem.svelte

```diff
@@ -1,18 +1,39 @@
 <script lang="ts">
-    import { watchlist, currentStock } from "$lib/stores";
+    import api from "$lib/api";
+    import {
+        defaultWatchlist,
+        userWatchlist,
+        currentStock,
+        user,
+    } from "$lib/stores";
     import type { Stock } from "$lib/types";
 
     export let stock: Stock;
     export let onClick: () => void;
 
+    const newRequest = () => api.user().watchlist_add({ ticker: stock.ticker });
+
     function click() {
-        watchlist.update((stocks: Stock[]) => {
-            if (stocks.indexOf(stock) === -1) {
-                return [...stocks, stock];
-            }
-            return stocks;
-        });
-        currentStock.set(stock);
+        if ($user) {
+            userWatchlist.update((tickers: string[]) => {
+                if (tickers.indexOf(stock.ticker) === -1) {
+                    newRequest();
+                    return [...tickers, stock.ticker];
+                }
+                return tickers;
+            });
+        } else {
+            defaultWatchlist.update((tickers: string[]) => {
```

Comment:
It might be helpful to have a single login component rooted somewhere in layout.ts, with a globally accessibly function that triggers it to show (perhaps asynchronous returning a bool indicating whether login succeeded, so that action can be taken after the user logs in?)